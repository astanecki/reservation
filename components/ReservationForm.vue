<template>
  <ReservationCard
    :price="price"
    :currency="currency"
    :rating="rating"
    :ratingQuantity="ratingQuantity"
  >
    <DatesInput
      @click="onInputClicked"
      @blur="onInputClicked(null)"
    />
    <Calendar
      :dateFrom="dateFrom"
      :dateTo="dateTo"
    />
  </ReservationCard>
</template>
<script>
import Calendar from "~/components/Calendar";
import ReservationCard from "~/components/ReservationCard";
import DatesInput from "~/components/DatesInput";

export default {
  name: 'ReservationForm',

  components: {
    ReservationCard,
    DatesInput,
    Calendar,
  },

  data() {
    return {
      chosenInputType: '',
    };
  },

  props: {
    price: {
      type: Number,
    },
    currency: {
      type: String,
    },
    rating: {
      type: Number,
    },
    dateFrom: {
      type: String,
    },
    dateTo: {
      type: String,
    },
    ratingQuantity: {
      type: Number,
    }
  },

  methods: {
    onInputClicked(type) {
      this.chosenInputType = type;
      console.log(`*********onInputClicked`, type);
    }
  }
}
</script>
