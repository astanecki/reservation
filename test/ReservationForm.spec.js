import { mount } from '@vue/test-utils'
import ReservationForm from '@/components/ReservationForm.vue'

// todo add to jest configuration not to add Vue.use in each case
import Vue from 'vue';
import VCalendar from 'v-calendar';

Vue.use(VCalendar);

describe('ReservationForm', () => {
  const propsData = {
    price: 298,
    currency: 'zÅ‚',
    rating: 4.5,
    ratingQuantity: 123,
    dateFrom: '2017-03-16',
    dateTo: '2017-03-20',
  };

  const createWrapper = () => {
    return mount(ReservationForm, {
      propsData,
    });
  };

  it('is a Vue instance', () => {
    const wrapper = createWrapper();

    expect(wrapper.vm).toBeTruthy();
  });

  it(`should match snapshot when props given`, () => {
    const wrapper = createWrapper();

    expect(wrapper.vm.$el).toMatchSnapshot();
  });
})
