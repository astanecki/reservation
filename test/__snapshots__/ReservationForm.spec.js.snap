// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationForm should match snapshot when props given 1`] = `
<div
  class="reservation-card"
>
  <label
    class="reservation-card__price"
  >
    298 zł
  </label>
   
  <div
    class="reservation-card__rating"
  >
    <div
      class="reservation-card__rating-stars"
    >
      <div
        class="stars"
      >
        <span
          class="stars--filled"
        >
          ★
        </span>
        <span
          class="stars--filled"
        >
          ★
        </span>
        <span
          class="stars--filled"
        >
          ★
        </span>
        <span
          class="stars--filled"
        >
          ★
        </span>
        ☆
      </div>
    </div>
     
    <div
      class="reservation-card__rating-quantity"
    >
      123
    </div>
  </div>
   
  <div
    class="reservation-card__divider"
  />
   
  <div
    class="dates-input"
  >
    <div
      class="dates-input__label"
    >
      
    Dates
  
    </div>
     
    <div
      class="dates-input__wrapper"
    >
      <input
        class=""
        readonly="readonly"
      />
       
      <span
        class="dates-input__arrow"
      >
        →
      </span>
       
      <input
        class=""
        readonly="readonly"
      />
    </div>
  </div>
   
  <!---->
  

  {}

  
  <button
    class="clear-button"
  >
    Clear
  </button>
</div>
`;
